src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java

```diff
@@ -123,31 +125,35 @@
 
   @Override
   public Future<Void> validate(MappingProfile mappingProfile, OkapiConnectionParams params) {
+    Promise<Void> promise = Promise.promise();
     if (CollectionUtils.isNotEmpty(mappingProfile.getTransformations())) {
-      return transformationFieldsService.getTransformationFields(params).compose(transformationFieldCollection -> {
-        List<TransformationField> transformationFields = transformationFieldCollection.getTransformationFields();
-        for (Transformations transformation : mappingProfile.getTransformations()) {
-          String fieldId = transformation.getFieldId();
-          if (StringUtils.isBlank(fieldId)) {
-            throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, "Field id is missing for mapping profile transformation");
-          }
-          Optional<TransformationField> transformationFieldOptional = transformationFields.stream()
-            .filter(transformationField -> fieldId.equals(transformationField.getFieldId()))
-            .findFirst();
-          if (transformationFieldOptional.isEmpty()) {
-            throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, String.format("Transformation doesn't exist by provided fieldId: %s", fieldId));
-          }
-          TransformationField transformationField = transformationFieldOptional.get();
-          TransformationField.RecordType expectedRecordType = transformationField.getRecordType();
-          if (Objects.isNull(transformation.getRecordType()) || !transformation.getRecordType().toString().equals(expectedRecordType.toString())) {
-            throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, String.format("Transformation record type is missing or incorrect according to provided fieldId: %s, " +
-              "expected record type: %s", fieldId, expectedRecordType));
-          }
-        }
-        return Future.succeededFuture();
-      });
+      transformationFieldsService.validateTransformations(mappingProfile.getTransformations())
+        .compose(v -> transformationFieldsService.getTransformationFields(params)
```

Comment:
why you use compose, and then onSuccess on tris compose, and on failure. Probably, it is better to use  onSuccess and on failure only if you use it.
Also, you forgot to complete promise if mappingProfile.getTransformations() is empty, that s why your tests are failed

Reply:
Fixed